src/main/java/com/clouway/asynctaskscheduler/gae/RoutingEventDispatcher.java

```diff
@@ -61,6 +60,31 @@ public void dispatchAsyncEvent(String eventClassAsString, String eventAsJson) th
     dispatchListeners(event);
   }
 
+  /**
+   * @param eventClassAsString
+   * @param eventAsJson
+   * @param listenerId
+   * @throws ClassNotFoundException
+   */
+  public void dispatchEventListener(String eventClassAsString, String eventAsJson, int listenerId) throws ClassNotFoundException {
+    if (Strings.isNullOrEmpty(eventClassAsString) || Strings.isNullOrEmpty(eventAsJson)) {
+      throw new IllegalArgumentException("No AsyncEvent class as string or evnt as json provided.");
+    }
+
+    Class<?> eventClass = Class.forName(eventClassAsString);
+
+    if (!Arrays.asList(eventClass.getInterfaces()).contains(AsyncEvent.class)) {
+      throw new IllegalArgumentException("No AsyncEvent class provided.");
+    }
+
+    AsyncEvent<AsyncEventHandler> event = getAsyncEvent(eventAsJson, eventClass);
+
+    List<? extends AsyncEventListener> listeners  = listenersFactory.create(event.getClass());
```

Comment:
this part is not well constructed, take a look in the dispatch "event handler" method